plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseSharedMetastoreTest.java

```diff
@@ -128,4 +134,57 @@ public void testShowSchemas()
                 showCreateIcebergWithRedirectionsSchema,
                 getExpectedIcebergCreateSchema("iceberg_with_redirections"));
     }
+
+    @Test
+    public void testTimeTravelAsOfVersionWithRedirection()
+    {
+        assertQuery("SELECT * FROM iceberg_with_redirections." + schema + ".nation", "SELECT * FROM nation");
+        long snapshot1 = getLatestSnapshotId();
+        long v1EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot1);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(25, 'POLAND', 3, 'test 1')", schema), 1);
+        long snapshot2 = getLatestSnapshotId();
+        long v2EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot2);
+        assertUpdate(format("INSERT INTO iceberg_with_redirections.%s.nation VALUES(26, 'CHILE', 1, 'test 2')", schema), 1);
+        long snapshot3 = getLatestSnapshotId();
+        long v3EpochMillis = getCommittedAtInEpochMilliSeconds(snapshot3);
+        long incorrectSnapshot = 2324324333L;
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d", schema, snapshot1), "SELECT * FROM nation");
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR TIMESTAMP AS OF %s", schema, timestampLiteral(v1EpochMillis)), "SELECT * FROM nation");
+        assertQuery(format("SELECT * FROM iceberg_with_redirections.%s.nation FOR VERSION AS OF %d WHERE nationkey = 25", schema, snapshot2), "VALUES(25, 'POLAND', 3, 'test 1')");
```

Comment:
What is the added value of the queries having `WHERE` clauses in this context?
We're checking the table redirection functionality and not the table filter functionality here.

Reply:
I just wanted to make sure that the row I inserted is there. But you are right that doesn't bring much value